appservice/src/createAppService/createWebApp.ts

```diff
@@ -23,28 +23,38 @@ export async function createWebApp(
 }
 
 export async function setWizardContextDefaults(wizardContext: IAppServiceWizardContext, actionContext: IActionContext, advancedCreation?: boolean): Promise<void> {
+    const isJavaProject: boolean = await javaUtils.isJavaProject();
+    if (isJavaProject) {
+        wizardContext.recommendedSiteRuntime = [
+            LinuxRuntimes.java,
+            LinuxRuntimes.tomcat,
+            LinuxRuntimes.wildfly
+        ];
+    }
     // only detect if one workspace is opened
     if (workspace.workspaceFolders && workspace.workspaceFolders.length === 1) {
         const fsPath: string = workspace.workspaceFolders[0].uri.fsPath;
         if (await fse.pathExists(path.join(fsPath, 'package.json'))) {
-            wizardContext.recommendedSiteRuntime = LinuxRuntimes.node;
+            wizardContext.recommendedSiteRuntime = [ LinuxRuntimes.node ];
         } else if (await fse.pathExists(path.join(fsPath, 'requirements.txt'))) {
             // requirements.txt are used to pip install so a good way to determine it's a Python app
-            wizardContext.recommendedSiteRuntime = LinuxRuntimes.python;
+            wizardContext.recommendedSiteRuntime = [ LinuxRuntimes.python ];
         }
-        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime;
+        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime ? [...wizardContext.recommendedSiteRuntime.keys()].join(',') : undefined;
```

Comment:
Simplify this line to work for an array instead of a map, but otherwise looks good

Reply:
Oh sorry, I missed this part. Updated...